apps/src/sites/studio/pages/levels/show.js

```diff
@@ -1,14 +1,20 @@
 import $ from 'jquery';
 import React from 'react';
 import ReactDOM from 'react-dom';
+import {getStore, registerReducers} from '@cdo/apps/redux';
 import ScriptLevelRedirectDialog from '@cdo/apps/code-studio/components/ScriptLevelRedirectDialog';
+import sectionData, {
+  setTtsAutoplayEnabled
+} from '@cdo/apps/redux/sectionDataRedux';
 
 $(document).ready(initPage);
 
 function initPage() {
   const script = document.querySelector('script[data-level]');
   const config = JSON.parse(script.dataset.level);
-
+  registerReducers({sectionData});
```

Comment:
Sorry for the basic questions, I'm still trying to figure out how the pieces fit together here.  I see that we now need section data on the level page (which wasn't present before, right?).  Where is setSection called from?  If a user is in more than one section, which section is set?

Reply:
No particular section is set overall but the redux section data `ttsAutoplayEnabled` is set. So for a student where it's enabled, the state of the store looks like,
<img width="354" alt="Screen Shot 2021-03-02 at 4 23 30 PM" src="https://user-images.githubusercontent.com/24235215/109733344-c6b2e380-7b73-11eb-9c37-b2285ee396ed.png">

Reply:
(there is no additional section data just that one field)

Reply:
@cforkish Any thoughts on this?  Something here feels off to me but I'm having trouble putting a finger on it or what an alternate design would look like.

Reply:
We could store it on the `state.currentUser` instead, even though it is part of the section data in the DB. I'm not sure if that would make it more or less confusing.

Reply:
@jamescodeorg the reason this feels off to me is that there is a confusing mix of rails and react in this code path. usually we setup the redux store in the same place where we render the components that consume the data we're setting up, whereas this `show.js` doesn't actually render anything unless there's a redirect url to handle. so it feels odd to be setting properties for components that aren't even referenced here. however, this appears to be the only common js entrypoint for the levels route, so imo this is the right place to put the redux setup.

as for where in redux we store this data, it also feels a bit off to be putting it in a part of the store which otherwise is not being used by this route, but conceptually i agree that `sectionData` is the right place. putting it in `userData` would feel off for all the same reason, _plus_ the fact that the property is defined at the section level.

lastly, part of the reason it looks off is just how much overhead is required to get this property into the component that needs it, but that's an architectural issue that is far out of scope for this work.

tl;dr this looks right to me, but i agree that it simultaneously looks off. hopefully once we start addressing our frontend architectural issues we'll be able to come up with nicer patterns for this kind of thing.

Reply:
it could be helpful to add a comment about putting it here because this is the common js entrypoint for the components that need it. also i'd add whitespace before and after these three lines to make it clearer that it's unrelated to the redirect dialog.